File: 0801-0900/0888-Fair Candy Swap/0888-Fair Candy Swap.py
class Solution:
    def fairCandySwap(self, A: List[int], B: List[int]) -> List[int]:
        diff = (sum(B) - sum(A)) // 2
        setB = set(B)
        for a in A:
            if a + diff in setB:
                return [a, a + diff]

File: 0801-0900/0888-Fair Candy Swap/0888-Fair Candy Swap.cpp
class Solution {
public:
    vector<int> fairCandySwap(vector<int>& A, vector<int>& B) {
        int diff = (accumulate(B.begin(), B.end(), 0) - accumulate(A.begin(), A.end(), 0)) / 2;
        unordered_set<int> setB(B.begin(), B.end());
        for (int a : A) {
            if (setB.count(a + diff)) {
                return {a, a + diff};
            }
        }
        return {};
    }
};
